lib/sensors/src/URM04Sensor.cpp

```diff
@@ -0,0 +1,211 @@
+#include "URM04Sensor.h"
+
+// instantiate pin connected to the URM04 URM04Sensor
+URM04Sensor::URM04Sensor::URM04Sensor(PinName trig_pin, PinName _RX, PinName _TX, uint8_t default_address)
+    : m_trigPin(trig_pin), startAddr(default_address), RX(_RX), TX(_TX), serial(TX, RX, BAUD_RATE) {
+  // instantiate command buffer with all zeros
+
+  memset(&cmdst[0], 0, sizeof(cmdst));
+}
+
+bool URM04Sensor::URM04Sensor::trigger_sensor() {
+  // turn on transmitting mode for RS485
+  m_trigPin.write(HIGH);
+  ThisThread::sleep_for(1ms);
+  /************ INSTANTIATE COMMANDS TO BE SENT OVER SERIAL*********/
+
+  // check sum represents the final bit in command buffer - made by adding all previous bits in command buffer
+  uint8_t checkSum;
+  // buffer header
+  cmdst[0] = 0x55;
+  cmdst[1] = 0xAA;
+  // device address
+  cmdst[2] = startAddr;
+  // length
+  cmdst[3] = 0x00;
+  // the command itself
+  cmdst[4] = 0x01;
+  // checksum bit
+  checkSum = cmdst[0] + cmdst[1] + cmdst[2] + cmdst[3] + cmdst[4];
+  // instantiate the last element in the command buffer with checksum value
+  cmdst[5] = checkSum;
```

Comment:
Just do `cmdst[5] = cmdst[0] + cmdst[1] + cmdst[2] + cmdst[3] + cmdst[4];` and remove the checkSum var. Same with the other checksum instances